File: fincapp-frontend/mobile-app/app/src/main/java/com/irwi/fincapp/models/WeightLog.java
package com.irwi.fincapp.models;

public class WeightLog {
    public long id;
    public String cloudId;
    public long animalId;
    public String animalCloudId;
    public double weightKg;
    public String logDate;
    public String syncStatus;

    public WeightLog(long id, String cloudId, long animalId, String animalCloudId, double weightKg, String logDate, String syncStatus) {
        this.id = id;
        this.cloudId = cloudId;
        this.animalId = animalId;
        this.animalCloudId = animalCloudId;
        this.weightKg = weightKg;
        this.logDate = logDate;
        this.syncStatus = syncStatus;
    }
}

File: fincapp-backend/src/FincApp.Core/WeightLog.cs
using System;

namespace FincApp.Core;

public class WeightLog
{
    public Guid Id { get; set; }
    public Guid AnimalId { get; set; }
    public decimal WeightKg { get; set; }
    public DateTime LogDate { get; set; }

    public Animal? Animal { get; set; }
}
